front_end/content.py

```diff
@@ -140,13 +140,41 @@ def get_role(self, user_id):
         else:
             return "student"
 
+    def get_users_from_role(self, role, course_id):
+        users = []
+        sql = '''SELECT user_id
+                 FROM permissions
+                 WHERE role = ?
+                 AND (course_id = ? OR course_id IS NULL)'''
+        rows = self.c.execute(sql, (role, course_id,))
+        for row in rows:
```

Comment:
This looks good. But I want to use this as an opportunity for you to practice using [list comprehensions](https://towardsdatascience.com/list-comprehensions-in-python-6fdc4748d268). Please change the last few lines of code to a single line of code that looks like something like this (you'll have to test it to make sure it works:

`return [row["user_id"] for row in rows]`

And remove the `users` variable.

List comprehensions are often faster than regular loops and use fewer lines of code.

Reply:
After that, I think we'll be good to go.